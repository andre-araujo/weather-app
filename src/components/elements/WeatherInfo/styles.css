.container {
  background: rgba(255, 255, 255, 0.8);
  padding: 10px 8px;
}

.largeText {
  font-size: 16px;
}

.smallText {
  font-size: 16px;
}
