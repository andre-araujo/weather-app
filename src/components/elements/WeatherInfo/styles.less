@import '../../../styles/settings/colors.less';

.container {
  padding: 0 8px;
  display: flex;
  height: 100px;
  overflow: hidden;
  transition: .2s height;
  border: 0;
  width: 100%;
  position: relative;
  box-shadow: 0 2px 6px rgba(0, 0, 0, .2)
}

.openInfo {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  width: 100%;
  cursor: pointer;
  background: 0;
  color: transparent;
  user-select: none;
  border: 0;
  z-index: 1;
}

.openInfo:disabled {
  pointer-events: none;
}

.content {
  color: @white;
  padding: 16px 0;
  font-weight: 400;
  text-align: left;
}

.temperature {
  margin-bottom: 30px;
}

.temperature time {
  display: block;
  margin-bottom: 15px;
  text-transform: uppercase;
  font-size: 20px;
}

.toggleTemperature,
.largeText {
  font-size: 23px;
  color: inherit;
  margin-bottom: 16px;
}

.toggleTemperature {
  padding: 0;
  background: none;
  border: 0;
  pointer-events: none;
  outline: none;
}

.smallText {
  font-size: 16px;
  margin-bottom: 4px;
  color: inherit;
}

.icon {
  font-size: 200px;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 16px 50px 0 0;
  width: 58%;
  flex:  0 0 58%;
  color: @white;
  text-align: center;
}

.hidden {
  opacity: 0;
  transition: .2s opacity;
}

.active {
  height: 280px;
  cursor: default;

  .hidden {
    opacity: 1;
  }

  .toggleTemperature {
    pointer-events: all;
  }

  .toggleTemperature:hover {
    cursor: pointer;
    text-shadow: 1px 1px @black10;
  }
}
