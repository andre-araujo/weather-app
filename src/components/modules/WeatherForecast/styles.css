.container {
  width: 100%;
  max-width: 340px;
  border: 1px solid black;
}
