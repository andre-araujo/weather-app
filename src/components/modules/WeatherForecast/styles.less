.container {
  width: 100%;
  max-width: auto;
  overflow: hidden;
  position: relative;
  z-index: 1;

  @media (min-width: 680px) {
    max-width: 540px;
  }
}

