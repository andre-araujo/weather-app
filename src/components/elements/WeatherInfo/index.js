import React from 'react';
import { string, number, bool, func, oneOfType } from 'prop-types';

import { CELCIUS, FAHRENHEIT } from '../../../lib/constants';
import WeatherIcon from '../WeatherIcon';
import styles from './styles.less';
import getTemperatureColor from '../../../lib/getTemperatureColor';
import fahrenheitToCelcius from '../../../lib/fahrenheitToCelcius';

function WeatherInfo({
  day,
  humidity,
  pressure,
  temperature,
  temperatureUnit,
  weather,
  weatherCode,
  wind,
  isOpen,
  onClick,
  toggleTemperatureUnit,
}) {
  const convertedTemperature =
    temperatureUnit === CELCIUS ? fahrenheitToCelcius(temperature) : temperature;

  return (
    <div
      className={`${styles.container} ${isOpen ? styles.active : ''}`}
      style={{ backgroundColor: getTemperatureColor(temperature) }}
    >
      {
        !isOpen && (
          <button
            className={styles.openInfo}
            disabled={isOpen}
            onClick={onClick}
            aria-label="Open Info"
            aria-required="true"
          >
            open info
          </button>
        )
      }
      <div className={`${styles.icon} ${styles.hidden}`}>
        <WeatherIcon weatherCode={weatherCode} />
      </div>
      <div className={styles.content}>
        <div className={styles.temperature}>
          <time dateTime={new Date()}>{day}</time>
          <button
            data-test="WeatherInfo_Temperature"
            disabled={!isOpen}
            className={styles.toggleTemperature}
            onClick={toggleTemperatureUnit}
          >
            {convertedTemperature}ยบ{temperatureUnit}
          </button>
        </div>
        <div className={styles.hidden}>
          {weather && <p className={styles.largeText}>{weather}</p>}
          {wind && <p className={styles.smallText}>Wind: {wind}</p>}
          {humidity && <p className={styles.smallText}>humidity: {humidity}</p>}
          {pressure && <p className={styles.smallText}>Pressure: {pressure}</p>}
        </div>
      </div>
    </div>
  );
}

WeatherInfo.defaultProps = {
  day: null,
  temperatureUnit: FAHRENHEIT,
  humidity: null,
  pressure: null,
  temperature: null,
  weather: null,
  weatherCode: null,
  wind: null,
  isOpen: false,
};

WeatherInfo.propTypes = {
  day: string,
  temperatureUnit: string,
  toggleTemperatureUnit: func,
  humidity: string,
  pressure: string,
  temperature: oneOfType([string, number]),
  weather: string,
  weatherCode: string,
  wind: string,
  isOpen: bool,
  onClick: func,
};

export default WeatherInfo;
