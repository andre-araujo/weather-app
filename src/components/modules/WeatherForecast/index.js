import React from 'react';

import LocationDisplay from '../../elements/LocationDisplay';
import WeatherInfo from '../../elements/WeatherInfo';
import cx from './styles.css';

function WeatherForecast() {
  return (
    <section className={cx.container}>
      <LocationDisplay city="Rio de Janeiro" state="Rio de Janeiro" />
      <WeatherInfo
        day="Hoje"
        humidity="78%"
        pressure="1003hPA"
        temperature={32}
        weather="Ensolarado"
        wind="NO 6.4Km/h"
      />
    </section>
  );
}

export default WeatherForecast;
