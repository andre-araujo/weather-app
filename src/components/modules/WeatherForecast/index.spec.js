import React from 'react';
import { shallow } from 'enzyme';

import WeatherForecast from './index';
import WeatherInfo from '../../elements/WeatherInfo';

jest.mock('../../../lib/getLatLong');

let props;

beforeEach(() => {
  props = {
    weather: {
      data: [
        {
          day: 'Today',
          humidity: '00',
          pressure: '00',
          temperature: '80',
          temperatureUnit: 'F',
          weather: '00',
          weatherCode: '00',
          wind: '00',
          onClick: jest.fn(),
          toggleTemperatureUnit: jest.fn(),
        },
        {
          day: 'Tomorrow',
          temperature: '80',
          temperatureUnit: 'F',
          weather: '00',
          weatherCode: '00',
          onClick: jest.fn(),
          toggleTemperatureUnit: jest.fn(),
        },
        {
          day: 'After Tomorrow',
          temperature: '80',
          temperatureUnit: 'F',
          weather: '00',
          weatherCode: '00',
          onClick: jest.fn(),
          toggleTemperatureUnit: jest.fn(),
        },
      ],
      location: {
        city: 'xx',
        region: 'xx',
      },
      units: {
        temperature: 'F',
      },
    },
  };
});

describe('WeatherForecast component', () => {
  it('should match snapshot with no error message', () => {
    expect(shallow(<WeatherForecast {...props} />)).toMatchSnapshot();
  });

  it('should set third item from infoStateList as true', () => {
    const wrapper = shallow(<WeatherForecast {...props} />);

    const itemProp = wrapper
      .find('li')
      .at(2)
      .find(WeatherInfo)
      .get(0)
      .props;

    expect(wrapper.state('infoStateList')).toEqual([true, false, false]);

    itemProp.onClick();

    expect(wrapper.state('infoStateList')).toEqual([false, false, true]);
  });

  it('should set second item from infoStateList as true', () => {
    const wrapper = shallow(<WeatherForecast {...props} />);

    const itemProp = wrapper
      .find('li')
      .at(1)
      .find(WeatherInfo)
      .get(0)
      .props;

    expect(wrapper.state('infoStateList')).toEqual([true, false, false]);

    itemProp.onClick();

    expect(wrapper.state('infoStateList')).toEqual([false, true, false]);
  });

  it('should toggleTemperatureUnit when WeatherInfo\'s toggleTemperatureUnit prop is called', () => {
    const wrapper = shallow(<WeatherForecast {...props} />);

    const itemProp = wrapper
      .find('li')
      .at(1)
      .find(WeatherInfo)
      .get(0)
      .props;

    expect(wrapper.state('temperatureUnit')).toEqual('F');

    itemProp.toggleTemperatureUnit();

    expect(wrapper.state('temperatureUnit')).toEqual('C');

    itemProp.toggleTemperatureUnit();

    expect(wrapper.state('temperatureUnit')).toEqual('F');
  });
});
