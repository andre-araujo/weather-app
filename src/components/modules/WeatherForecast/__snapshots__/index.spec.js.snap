// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherForecast component should match snapshot with no error message 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgba(255, 202, 102, 0.3)",
    }
  }
>
  <section>
    <LocationDisplay
      city="xx"
      state="xx"
    />
    <ul>
      <li
        key="Today"
      >
        <WeatherInfo
          day="Today"
          humidity="00"
          isOpen={true}
          onClick={[Function]}
          pressure="00"
          temperature="80"
          temperatureUnit="F"
          toggleTemperatureUnit={[Function]}
          weather="00"
          weatherCode="00"
          wind="00"
        />
      </li>
      <li
        key="Tomorrow"
      >
        <WeatherInfo
          day="Tomorrow"
          humidity={null}
          isOpen={false}
          onClick={[Function]}
          pressure={null}
          temperature="80"
          temperatureUnit="F"
          toggleTemperatureUnit={[Function]}
          weather="00"
          weatherCode="00"
          wind={null}
        />
      </li>
      <li
        key="After Tomorrow"
      >
        <WeatherInfo
          day="After Tomorrow"
          humidity={null}
          isOpen={false}
          onClick={[Function]}
          pressure={null}
          temperature="80"
          temperatureUnit="F"
          toggleTemperatureUnit={[Function]}
          weather="00"
          weatherCode="00"
          wind={null}
        />
      </li>
    </ul>
  </section>
</div>
`;
